lib/engine-object-file/src/artifact.rs

```diff
@@ -384,12 +384,12 @@ impl ObjectFileArtifact {
                 .into_boxed_slice(),
             signatures: signatures.into_boxed_slice(),
             metadata_length: 0,
-            symbol_registry,
+            symbol_registry: Arc::new(symbol_registry),
         })
     }
 
-    /// Get the `SymbolRegistry` used to generate the names used in the Artifact.
-    pub fn symbol_registry(&self) -> &dyn SymbolRegistry {
+    /// Get the `SymbolRegistry` used to generate the names used in the artifact.
+    pub fn symbol_registry(&self) -> &Arc<ModuleMetadataSymbolRegistry> {
```

Comment:
Technically this is a breaking change in the public API of `engine-object-file`.

Not sure what best practice is here for our own, essentially internal APIs. It is unfortunate that this is at least theoretically fully usable on crates.io and we'd be breaking it.